spec/models/user_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    it 'is valid with a name and non-negative post counter' do
+      user = User.new(Name: 'John Doe', PostsCounter: 0)
+      expect(user).to be_valid
+    end
+
+    it 'is invalid without a name' do
+      user = User.new(Name: nil, PostsCounter: 0)
+      expect(user).not_to be_valid
+    end
+
+    it 'is invalid with a negative post counter' do
+      user = User.new(Name: 'John Doe', PostsCounter: -1)
+      expect(user).not_to be_valid
+    end
+  end
+
+  describe 'associations' do
+    it 'has many posts' do
+      association = User.reflect_on_association(:posts)
+      expect(association.macro).to eq(:has_many)
+    end
+
+    it 'has many comments' do
+      association = User.reflect_on_association(:comments)
+      expect(association.macro).to eq(:has_many)
+    end
+
+    it 'has many likes' do
+      association = User.reflect_on_association(:likes)
+      expect(association.macro).to eq(:has_many)
+    end
+  end
+
+  describe 'recent_post' do
+    let(:user) { User.create(Name: 'John Doe', PostsCounter: 0) }
+
+    it 'returns the most recent posts in descending order' do
+      post1 = user.posts.create(title: 'First post', content: 'Lorem ipsum')
```

Comment:
- [ ] some of your tests are failing, kindly fix them
![image](https://github.com/KUBAHO3/blog-app/assets/66097371/8d3e8421-e143-4043-965f-d9113ecb8eb5)